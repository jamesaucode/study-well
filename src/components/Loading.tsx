import React from "react";
import styled, { keyframes } from "styled-components";

const rotate = keyframes`
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
`;

const Spinner = styled.div`
  position: relative;
  margin: 0 auto;
  border-radius: 50%;
  width: 2em;
  height: 2em;
  overflow: hidden;
  border-top: 0.35em solid rgba(0, 0, 0, 0.2);
  border-right: 0.35em solid rgba(0, 0, 0, 0.2);
  border-bottom: 0.35em solid rgba(0, 0, 0, 0.2);
  border-left: 0.35em solid #3140f1;
  animation: ${rotate} 1s linear infinite;
  text-indent: -9999em;
`;
interface BouncingDotProps {
  delay: number;
}
const bouncing = keyframes`
  0% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(10px);
  }
  100% {
    transform: translateY(0);
  }
`;
const expanding = keyframes`
  0% {
    transform: scale(0.5);
  }
  50% {
    transform: scale(1);
  }
  100% {
    transform: scale(0.5);
  }
`;
const BouncingDot = styled.circle<BouncingDotProps>`
  border-radius: 50%;
  height: 8px;
  width: 8px;
  background-color: #999;
  fill: #999;
  /* animation: ${bouncing} 1000ms infinite; */
  animation: ${expanding} 1000ms infinite;
  animation-delay: ${(props): number => props.delay}ms;
`;
const Wrapper = styled.div`
  display: flex;
  margin: 0 auto;
  position: relative;
`;
const SVG = styled.svg`
  height: 10px;
  width: 10px;
  margin: 0 3px;
`;
const Loading: React.FunctionComponent = (): JSX.Element => {
  return (
    <React.Fragment>
      {/* <Spinner /> */}
      <Wrapper>
        <SVG height={10} width={10} viewBox="0 0 20 20">
          <BouncingDot delay={200} cx="10" cy="10" r="10" />
        </SVG>
        <SVG height={10} width={10} viewBox="0 0 20 20">
          <BouncingDot delay={400} cx="10" cy="10" r="10" />
        </SVG>
        <SVG height={10} width={10} viewBox="0 0 20 20">
          <BouncingDot delay={600} cx="10" cy="10" r="10" />
        </SVG>
      </Wrapper>
    </React.Fragment>
  );
};

export default Loading;
