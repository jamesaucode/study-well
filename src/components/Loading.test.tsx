/* eslint-disable */
import { mount } from "enzyme";
import Loading from "./Loading";

describe("Loading", () => {
  it("Component renders", () => {
    const wrapper = mount(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
