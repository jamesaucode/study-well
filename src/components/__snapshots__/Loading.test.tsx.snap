// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders 1`] = `
<Loading>
  <Loading__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Loading__Wrapper-s1l57m-2",
            "isStatic": true,
            "lastClassName": "icyLHk",
            "rules": Array [
              "display:flex;margin:0 auto;position:relative;",
            ],
          },
          "displayName": "Loading__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Loading__Wrapper-s1l57m-2",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Loading__Wrapper-s1l57m-2 icyLHk"
      >
        <Loading__SVG
          height={10}
          viewBox="0 0 20 20"
          width={10}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Loading__SVG-s1l57m-3",
                  "isStatic": true,
                  "lastClassName": "chkjnV",
                  "rules": Array [
                    "height:10px;width:10px;margin:0 3px;",
                  ],
                },
                "displayName": "Loading__SVG",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Loading__SVG-s1l57m-3",
                "target": "svg",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height={10}
            viewBox="0 0 20 20"
            width={10}
          >
            <svg
              className="Loading__SVG-s1l57m-3 chkjnV"
              height={10}
              viewBox="0 0 20 20"
              width={10}
            >
              <Loading__BouncingDot
                cx="10"
                cy="10"
                delay={200}
                r="10"
              >
                <StyledComponent
                  cx="10"
                  cy="10"
                  delay={200}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Loading__BouncingDot-s1l57m-1",
                        "isStatic": false,
                        "lastClassName": "jzXCHq",
                        "rules": Array [
                          "border-radius:50%;height:8px;width:8px;background-color:#999;fill:#999;animation:",
                          Keyframes {
                            "id": "sc-keyframes-dTRBUS",
                            "inject": [Function],
                            "name": "dTRBUS",
                            "rules": Array [
                              "@-webkit-keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                              "@keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                            ],
                            "toString": [Function],
                          },
                          " 1000ms infinite;animation-delay:",
                          [Function],
                          "ms;",
                        ],
                      },
                      "displayName": "Loading__BouncingDot",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Loading__BouncingDot-s1l57m-1",
                      "target": "circle",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  r="10"
                >
                  <circle
                    className="Loading__BouncingDot-s1l57m-1 fppocI"
                    cx="10"
                    cy="10"
                    r="10"
                  />
                </StyledComponent>
              </Loading__BouncingDot>
            </svg>
          </StyledComponent>
        </Loading__SVG>
        <Loading__SVG
          height={10}
          viewBox="0 0 20 20"
          width={10}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Loading__SVG-s1l57m-3",
                  "isStatic": true,
                  "lastClassName": "chkjnV",
                  "rules": Array [
                    "height:10px;width:10px;margin:0 3px;",
                  ],
                },
                "displayName": "Loading__SVG",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Loading__SVG-s1l57m-3",
                "target": "svg",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height={10}
            viewBox="0 0 20 20"
            width={10}
          >
            <svg
              className="Loading__SVG-s1l57m-3 chkjnV"
              height={10}
              viewBox="0 0 20 20"
              width={10}
            >
              <Loading__BouncingDot
                cx="10"
                cy="10"
                delay={400}
                r="10"
              >
                <StyledComponent
                  cx="10"
                  cy="10"
                  delay={400}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Loading__BouncingDot-s1l57m-1",
                        "isStatic": false,
                        "lastClassName": "jzXCHq",
                        "rules": Array [
                          "border-radius:50%;height:8px;width:8px;background-color:#999;fill:#999;animation:",
                          Keyframes {
                            "id": "sc-keyframes-dTRBUS",
                            "inject": [Function],
                            "name": "dTRBUS",
                            "rules": Array [
                              "@-webkit-keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                              "@keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                            ],
                            "toString": [Function],
                          },
                          " 1000ms infinite;animation-delay:",
                          [Function],
                          "ms;",
                        ],
                      },
                      "displayName": "Loading__BouncingDot",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Loading__BouncingDot-s1l57m-1",
                      "target": "circle",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  r="10"
                >
                  <circle
                    className="Loading__BouncingDot-s1l57m-1 iBYToF"
                    cx="10"
                    cy="10"
                    r="10"
                  />
                </StyledComponent>
              </Loading__BouncingDot>
            </svg>
          </StyledComponent>
        </Loading__SVG>
        <Loading__SVG
          height={10}
          viewBox="0 0 20 20"
          width={10}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Loading__SVG-s1l57m-3",
                  "isStatic": true,
                  "lastClassName": "chkjnV",
                  "rules": Array [
                    "height:10px;width:10px;margin:0 3px;",
                  ],
                },
                "displayName": "Loading__SVG",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Loading__SVG-s1l57m-3",
                "target": "svg",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height={10}
            viewBox="0 0 20 20"
            width={10}
          >
            <svg
              className="Loading__SVG-s1l57m-3 chkjnV"
              height={10}
              viewBox="0 0 20 20"
              width={10}
            >
              <Loading__BouncingDot
                cx="10"
                cy="10"
                delay={600}
                r="10"
              >
                <StyledComponent
                  cx="10"
                  cy="10"
                  delay={600}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Loading__BouncingDot-s1l57m-1",
                        "isStatic": false,
                        "lastClassName": "jzXCHq",
                        "rules": Array [
                          "border-radius:50%;height:8px;width:8px;background-color:#999;fill:#999;animation:",
                          Keyframes {
                            "id": "sc-keyframes-dTRBUS",
                            "inject": [Function],
                            "name": "dTRBUS",
                            "rules": Array [
                              "@-webkit-keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                              "@keyframes dTRBUS{0%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}50%{-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}100%{-webkit-transform:scale(0.5);-ms-transform:scale(0.5);transform:scale(0.5);}}",
                            ],
                            "toString": [Function],
                          },
                          " 1000ms infinite;animation-delay:",
                          [Function],
                          "ms;",
                        ],
                      },
                      "displayName": "Loading__BouncingDot",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Loading__BouncingDot-s1l57m-1",
                      "target": "circle",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  r="10"
                >
                  <circle
                    className="Loading__BouncingDot-s1l57m-1 jzXCHq"
                    cx="10"
                    cy="10"
                    r="10"
                  />
                </StyledComponent>
              </Loading__BouncingDot>
            </svg>
          </StyledComponent>
        </Loading__SVG>
      </div>
    </StyledComponent>
  </Loading__Wrapper>
</Loading>
`;
